examples/02_miscellaneous/plot_1_pearson_correlation.py

```diff
@@ -0,0 +1,219 @@
+from fugw.utils import _make_tensor
+
+# %%
+# We first fetch 5 contrasts for each subject from the localizer dataset.
+n_subjects = 2
+
+contrasts = [
+    "sentence reading vs checkerboard",
+    "sentence listening",
+    "calculation vs sentences",
+    "left vs right button press",
+    "checkerboard",
+]
+
+brain_data = datasets.fetch_localizer_contrasts(
+    contrasts,
+    n_subjects=n_subjects,
+    get_anats=True,
+    get_masks=True,
+)
+
+source_imgs_paths = brain_data["cmaps"][0 : len(contrasts)]
+target_imgs_paths = brain_data["cmaps"][len(contrasts) : 2 * len(contrasts)]
+source_mask = brain_data["masks"][0]
+
+source_im = image.load_img(source_imgs_paths)
+target_im = image.load_img(target_imgs_paths)
+mask = image.load_img(source_mask)
+
+# %%
+# We then downsample the images by 5 to reduce the computational cost.
+SCALE_FACTOR = 5
+
+resized_source_affine = source_im.affine.copy() * SCALE_FACTOR
+resized_target_affine = target_im.affine.copy() * SCALE_FACTOR
+
+source_im_resized = image.resample_img(source_im, resized_source_affine)
+target_im_resized = image.resample_img(target_im, resized_target_affine)
+mask_resized = image.resample_img(mask, resized_source_affine)
+
+source_maps = np.nan_to_num(source_im_resized.get_fdata())
+target_maps = np.nan_to_num(target_im_resized.get_fdata())
+segmentation = mask_resized.get_fdata()
+
+coordinates = np.argwhere(segmentation > 0)
+
+source_features = source_maps[
+    coordinates[:, 0], coordinates[:, 1], coordinates[:, 2]
+].T
+target_features = target_maps[
+    coordinates[:, 0], coordinates[:, 1], coordinates[:, 2]
+].T
+
+
+# %%
+# We then compute the distance matrix between voxel coordinates.
+source_geometry = distance_matrix(coordinates, coordinates)
+target_geometry = source_geometry.copy()
+
+# %%
+# In order to avoid numerical errors when fitting the mapping, we normalize
+# all feature and geometry arrays.
+source_features_normalized = source_features / np.linalg.norm(
+    source_features, axis=1
+).reshape(-1, 1)
+target_features_normalized = target_features / np.linalg.norm(
+    target_features, axis=1
+).reshape(-1, 1)
+source_geometry_normalized = source_geometry / np.max(source_geometry)
+target_geometry_normalized = target_geometry / np.max(target_geometry)
+
+
+# %%
+# We first define a function to compute the Pearson correlation
+# between two tensors. Such function is not available in PyTorch,
+# but it is easy to implement.
+def pearson_corr(x, y, plan):
+    """
+    Compute the Pearson correlation between transformed
+    source features and target features.
+    """
+    if not torch.is_tensor(x) or not torch.is_tensor(y):
+        x = torch.tensor(x).to(torch.float32)
+        y = torch.tensor(y).to(torch.float32)
+
+    # Compute the transformed features
+    x_transformed = (
+        (plan @ x.T / plan.sum(dim=1).reshape(-1, 1)).T.detach().cpu()
+    )
+
+    vx = x_transformed - torch.mean(x_transformed)
+    vy = y - torch.mean(y)
```

Comment:
Actually, `torch.mean(x_transformed)` is not what you want here: you need to compute the mean along a specific axis.